model.sample.json

```diff
@@ -0,0 +1,274 @@
+                "label": {
+                    "short": "Kids",
+                    "medium": "Children"
+                },
+                "fields": ["Kids"]
+            }
+        ]
+    },
+    {
+        "name": "Person",
+        "label": "Person",
+        "collectionLabel": "People",
+        "fields": [
+            {
+                "name": "Id",
+                "type": "integer",
+                "required": true,
+                "initial_value": 0,
+                "key_field": true
+            },
+            {
+                "name": "Birthday",
+                "label": {
+                    "short": "DOB",
+                    "medium": "Birthday",
+                    "long": "Date of Birth"
+                },
+                "type": "Date",					
+                "placeholder": "Date of Birth",
+                "required": true,
+                "hints": {
+                    "width": 120
+                }
+            },
+            {
+                "name": "Title",
+                "label": {
+                    "short": "Title",
+                    "medium": "First Name"
+                },
+                "type": "text",
+                "placeholder": "First Name"
+            },
+            {
+                "name": "Firstname",
+                "label": {
+                    "short": "First",
+                    "medium": "First Name"
+                },
+                "type": "text",
+                "placeholder": "First Name",
+                "required": true
+            },
+            {
+                "name": "Middlename",
+                "label": {
+                    "short": "Middle",
+                    "medium": "Middle Name"
+                },
+                "type": "text",
+                "placeholder": "Middle Name"
+            },
+            {
+                "name": "MiddleInitial",
+                "label": {
+                    "short": "M. I.",
+                    "medium": "Middle Initial"
+                },
+                "type": "derived",
+                "placeholder": "Middle Initial",
+                "pattern": "{substring(Middlename, 0, 1)}}"
+            },
+            {
+                "name": "Lastname",
+                "label": {
+                    "short": "Last",
+                    "medium": "Last Name"
+                },
+                "type": "text",
+                "placeholder": "Last Name",
+                "required": true
+            },
+            {
+                "name": "Suffix",
+                "label": {
+                    "short": "Suf",
+                    "medium": "Suffix"
+                },
+                "type": "text",
+                "placeholder": "Suffix"
+            },
+            {
+                "name": "Fullname",
+                "label": "Full Name",
+                "type": "derived",
+                "pattern": "{Title }{Firstname}{ Middlename}{ Lastname}{ Suffix}",
```

Comment:
I like the spaces inside the `{}` as a way to make the spacing conditional.  Good idea!

Reply:
I don't like the spaces inside the {} because they are not consistent.  I think you should use a space after the opening { and a space before the closing }.

Reply:
The spacing does look a little weird, but the idea was that the spaces would be included in the output, so that you didn't have to include additional logic to include the spaces when concatenating the values together.  This is a little limited though, since punctuation and other values would not work.  What if we included additional curly brackets in the value that indicated formatting.  Something like:
`{Title{ }}{Firstname}{{ }Middlename}{{ }Lastname}{{ }Suffix}`
or
`{Address1{\n}}{{Address2}{\n}}{City{, }}{State{ }}{Zip}`
Where the value in the embedded curly brackets are only included if the evaluated expression in the outer curly brackets is truthy. Then values outside of the curly brackets would be considered constants.

Reply:
What do you guys think of my suggestion for the pattern expression?
@JimPatterson @KatFrog

Reply:
I'm feel a little mixed.  I see the idea and it makes sense, but I can't think of a system doing anything similar.  What about using an existing language like handlebars or mustache?  The downside I see with those is that they are much more verbose.  The upside is they are much more powerful.

I like the brevity of what you have but not the syntax itself.  So maybe just a tweak to the syntax.  How about using quotes around the literal part, and an operator to indicate what to do with the literal.  I could then see & (and) and | (or) as the operators.

So {Title&" "}{Name|"user"}

That would leave room for future expansion with other operators, like conditional blocks formed using a ternary operator syntax like {city&" "}{state?", "{state}}.

The use of quotes around the literals lets the user add white spaces to make it readable like {city & " "}{state ? ", "{state}}

Reply:
I like the direction.  I think we will just need to clarify the "Rules" later.  We'll probably have to use single quotes unless we want to escape all of the double quotes.

Reply:
Sounds like a plan to me.